### PR TITLE
Replace valid with preview and viewer

### LINKED ISSUE #1446: Add fields `viewer` and `preview` to /is-valid
For coherence with /valid, we should add the `viewer` and `preview` fields to /is-valid.

We should also consider deprecating the current `valid` field (as in https://github.com/huggingface/datasets-server/issues/1445). Note that it's in use in https://github.com/search?q=org%3Ahuggingface+datasets-server.huggingface.co+repo%3Ahuggingface%2Fnotebooks&type=code and also in the @lewtun's evaluator if I remember correctly.

### PR DESCRIPTION
**Pull request by severo**
replaces #1450 and #1447

fixes #1446 and #1445

- [x] remove `valid` field from `/valid` endpoint
- [x] replace `valid` with `viewer` and `preview` in `/is-valid`
- [x] update the docs
- [x] update openapi, <strike>rapidapi</strike> (I don't really understand rapidapi anymore), postman

This change is breaking the API for /valid and /is-valid. See https://moon-ci-docs.huggingface.co/docs/datasets-server/pr_1452/en/valid.

- [x] for Autotrain, I think the endpoints are not used in the code (https://github.com/huggingface/autotrain-ui and https://github.com/huggingface/autotrain-backend/). cc @abhishekkrthakur 
- [x] for model-evaluator, I opened https://github.com/huggingface/model-evaluator/pull/71 cc @lewtun

### CODE REVIEW

File: docs/source/valid.mdx [hunk 1]
```diff
@@ -2,17 +2,18 @@
...
-  <a href="https://huggingface.co/docs/datasets/stream">streamable</a> datasets
-  are supported so Datasets Server can extract the 100 first rows without downloading the
-  whole dataset. This is especially useful for previewing large datasets where downloading
-  the whole dataset may take hours!
+  The largest datasets are partially supported by Datasets Server. If they are{" "}
+  <a href="https://huggingface.co/docs/datasets/stream">streamable</a>, it can
+  extract the 100 first rows without downloading the whole dataset. This is
+  especially useful for previewing large datasets where downloading the whole
+  dataset may take hours! See the <code>preview</code> field in the response of{" "}
+  <code>/valid</code>.
```
Comment (stevhliu) [inline docs/source/valid.mdx:16]:
```suggestion
  <a href="https://huggingface.co/docs/datasets/stream">streamable</a>, Datasets Server can
  extract the first 100 rows without downloading the whole dataset. This is
  especially useful for previewing large datasets where downloading the whole
  dataset may take hours! See the <code>preview</code> field in the response of{" "}
  <code>/valid</code> to view these partially supported datasets.
```

File: docs/source/valid.mdx [hunk 2]
```diff
@@ -58,16 +59,27 @@ curl https://datasets-server.huggingface.co/valid \
 </curl>
 </inferencesnippet>
 
-The endpoint response is a JSON containing a list valid datasets nested under the `valid` key:
+The endpoint response is a JSON containing lists of datasets nested under the following keys:
+
+- `viewer`: the dataset is fully-supported and the Dataset Viewer is working on the Hub dataset page. It also means that the dataset has been auto-converted to Parquet.
```
Comment (stevhliu) [inline docs/source/valid.mdx:64]:
```suggestion
- `viewer`: the dataset is fully-supported and the Dataset Viewer is working on the Hub dataset page. It also means the dataset has been auto-converted to Parquet.
```

File: docs/source/valid.mdx [hunk 3]
```diff
@@ -58,16 +59,27 @@ curl https://datasets-server.huggingface.co/valid \
...
+- `preview`: the dataset is partially supported. The Dataset Viewer on the Hub shows a Preview of the first 100 rows, obtained by streaming.
```
Comment (stevhliu) [inline docs/source/valid.mdx:65]:
```suggestion
- `preview`: the dataset is partially supported and the Dataset Viewer on the Hub dataset page shows a preview of the first 100 rows obtained by streaming.
```

**Review by stevhliu — APPROVED** [review]
Cool update to `/valid`!

File: services/worker/src/worker/job_runners/dataset/is_valid.py [hunk 4]
```diff
@@ -5,38 +5,44 @@
...
     Args:
         dataset (`str`):
             A namespace (user or an organization) and a repo name separated
             by a `/`.
+        processing_graph (`ProcessingGraph`):
+            The processing graph. In particular, it must provide the list of
+            processing steps that enable the viewer and the preview.
     Returns:
-        `DatasetIsValidResponse`: An object with the is_valid_response.
+        `DatasetIsValidResponse`: The response (viewer, preview, valid).
```
Comment (AndreaFrancis) [inline services/worker/src/worker/job_runners/dataset/is_valid.py:34]: Not quite understand, but should `valid` be removed from here? 
```suggestion
        `DatasetIsValidResponse`: The response (viewer, preview).
```

File: services/worker/src/worker/job_runners/dataset/is_valid.py [hunk 5]
```diff
@@ -48,8 +54,24 @@ def get_job_type() -> str:
...
+        processing_graph: ProcessingGraph,
+    ) -> None:
+        super().__init__(
+            job_info=job_info,
+            app_config=app_config,
+            processing_step=processing_step,
+        )
+        self.processing_graph = processing_graph
+
     def compute(self) -> JobResult:
```
Comment (AndreaFrancis) [inline services/worker/src/worker/job_runners/dataset/is_valid.py:71]: Maybe you can use `CompleteJobResult` and change `compute_is_valid_response` return type to `DatasetIsValidResponse`.
So then, you return here CompleteJobResult(compute_is_valid_response....)
You don't need 1.0 anymore if it is a CompleteJobResult.

**Review by AndreaFrancis — APPROVED** [review]

File: services/worker/src/worker/job_runners/dataset/is_valid.py (hunk 5 above)
Comment (severo) [reply services/worker/src/worker/job_runners/dataset/is_valid.py:71]: Right! I'll change it

Comment (severo) [reply services/worker/src/worker/job_runners/dataset/is_valid.py:70]: Done, thanks for the idea!

**Pull request merged**